File: algs4/search/test/2019-04-29/bubble.c
#include <stdio.h>

void display(int* arr, int len){
	int i = 0;
	for(i=0; i<len; i++)
		printf("[%d].", arr[i]);
	printf("\b \n");
}

void swap(int *a, int *b){
	int tmp = *a;
	*a = *b;
	*b = tmp;
}

void bubblesort(int* arr, int len){
	int i, j;
	for(i=0; i<len; i++){
		for(j=1; j<len; j++){
			if(arr[j] < arr[j-1]){
				swap(&arr[j], &arr[j-1]);
			}
		}
	}
}

void main(){
	int arr[5] = {2,3,1,5,4};
	bubblesort(arr, 5);
	display(arr, 5);
}

File: go/sort/bubble.go
package main

import  "fmt"

func display(arr [5]int) {
	for _,data := range arr {
		fmt.Printf("%d\n",data)
	}
}

func main() {
	var arr [5]int = [5]int{3,4,2,5,1}
	display(arr)
}
